### ISSUE TITLE
nested hash with mixed string/symbol keys breaks parsing

### ISSUE THREAD

**Issue by 12eh** [seattlerb/ruby_parser#209]
Seems similar to https://github.com/seattlerb/ruby_parser/issues/182 except this is the code that causes the error:
```
  belongs_to :requester, class_name: 'User'
  foo = []
  foo << { 'foo': { bar: { baz: 'quux' } } }
```

Ran into this via brakeman, but reproducible with just RubyParser:
```
[35] pry(main)> RubyParser.new.parse "belongs_to :requester, class_name: 'User'; foo = []; foo << { 'foo': { bar: { baz: 'quux' } } }"
Racc::ParseError: (string):1 :: parse error on value ":" (tCOLON)
from /Users/josh/.rbenv/versions/2.3.0/lib/ruby/2.3.0/racc/parser.rb:528:in `on_error'
```

Replace the key `'foo'` with `foo`, and there's no error.

**Comment by presidentbeef** [comment]
This is the same as #175 and should be fixed by #208.

**Comment by zenspider** [comment]
This is fixed.
